### ISSUE TITLE
onSnapToItem is not fired when difference between sliderWidth and itemWidth is not much. 

### ISSUE THREAD

**Issue by hansen94** [archriss/react-native-snap-carousel#268]
### Is this a bug report or a feature request?

bug report

### Have you read [the guidelines](https://github.com/archriss/react-native-snap-carousel/blob/master/CONTRIBUTING.md) regarding bug report?

yes

### Have you read [the documentation](https://github.com/archriss/react-native-snap-carousel/blob/master/README.md) in its entirety?

yes

### Have you made sure that your issue hasn't already been reported/solved?

yes

### Is the bug specific to iOS or Android? Or can it be reproduced on both platforms?

(Write your answer here and specify the iOS/Android versions on which you've been able to reproduce the issue.)
specific to Android. Tested on Pixel XL API 27. 
Worked fine on ios. 

### Is the bug reproductible in a production environment (not a debug one)?

yes

### Have you been able to reproduce the bug in [the provided example](https://github.com/archriss/react-native-snap-carousel/tree/master/example)?

no, cannot run example for some reason. But i did try it on a fresh project and the bug occur.

### Environment

Environment:
    "react": "16.2.0",
    "react-native": "0.53.0",
    "react-native-snap-carousel": "^3.6.0"
Target Platform: 
    Android 6.0

### Steps to Reproduce

I created a project at https://github.com/hansen94/react-native-snap-carousel-test.git

Step to reproduce:
  1. Run project in Android in debug JS Remotely mode

You'll see that onSnapToItem does not fire even though visually, the carousel snaps. 

i noticed that onSnapToItem will fire if i reduce itemWidth or width in slideStyle, but i would prefer not to do that as it would not align the item to the center.

Basically, I want to have the next item peeks out from the right.

### Expected Behavior

onSnapToItem should fire if carousel snap regardless of item width

### Actual Behavior

onSnapToItem does not fire. In the sample code, I added console.log in onSnapToItem but it logs nothing when carousel snaps.

### Reproducible Demo

(Paste the link to an example project or paste the entirety of the relevant source code. Then provide instructions to reproduce the issue.)
https://github.com/hansen94/react-native-snap-carousel-test.git

**Comment by bd-arc** [comment]
Hi @hansen94,

Would you mind recreating on [Snack](https://snack.expo.io/) a minimal example that reproduces the problem ?

I'll gladly take a look at your issue, but having a Snack project at hand will save me a lot of time ;-)

**Comment by hansen94** [comment]
Sure. Here's the link: https://snack.expo.io/SJy6EP9DM

In the example, the number suppose to change when onSnapToItem is fired.
If you change the difference between sliderWidth and itemWidth to be greater than 50, it will work fine.
otherwise, onSnapToItem is not fired.

**Comment by bd-arc** [comment]
@hansen94 Thanks for the Snack!

I can't seem to reproduce the issue. Here is an updated Snack in which I've made sure to use the same `itemWidth` value everywhere: https://snack.expo.io/SkNX1hqDf

Can you reproduce the issue in this one? I'll try on other devices this afternoon.

**Comment by hansen94** [comment]
Here's the updated code: https://snack.expo.io/rkV-CnqDz

In here, I made a new const SLIDE_WIDTH that is smaller than ITEM_WIDTH to make the next slide "peeks out". onsnaptoitem is not fired in the transition to the last slide only. If I made the data array to have only 2 entries, it does not fire at all.
I tested this on Samsung S8 with Android 7.0
Are you able to reproduce this?

Here's some screenshot to show it: 
![screenshot_20180221-164413](https://user-images.githubusercontent.com/8460050/36473490-3bde76d8-1727-11e8-8f76-515b6ca8bf70.png)
![screenshot_20180221-164420](https://user-images.githubusercontent.com/8460050/36473491-3c224de0-1727-11e8-90f0-2c8e387567ea.png)
![screenshot_20180221-164426](https://user-images.githubusercontent.com/8460050/36473493-3c5c027e-1727-11e8-875e-93ea65a407d4.png)
![screenshot_20180221-164431](https://user-images.githubusercontent.com/8460050/36473494-3c9137dc-1727-11e8-9842-1c8cc993fe09.png)
![screenshot_20180221-164438](https://user-images.githubusercontent.com/8460050/36473510-480baba6-1727-11e8-89a3-a84c51563e0d.png)


Maybe i am doing this incorrectly. Is there a "correct" way to make the next slide "peeks out" in react-native-snap-carousel?

**Comment by bd-arc** [comment]
@hansen94 That's clearly not the proper way to get "peeking" items ;-) In fact, you're messing with the carousel logic, hence your issue.

Forget about using a different value for `slideStyle: { width }` and `itemWidth`. You should just play with [prop `inactiveSlideScale`](https://github.com/archriss/react-native-snap-carousel/blob/master/doc/PROPS_METHODS_AND_GETTERS.md#style-and-animation) to apply a less important scale factor than the default one (use something like `0.97` for example). Or you can set it to `1` and apply [a custom margin](https://github.com/archriss/react-native-snap-carousel/blob/master/doc/TIPS_AND_TRICKS.md#margin-between-slides).

**Comment by hansen94** [comment]
Oh i see. Thank you for you guidance. inactiveSlideScale does what i want. 
Here's the final code that i use:
`import React, { Component } from 'react';
import { Text, View, Dimensions } from 'react-native';

// You can import from local files
// import AssetExample from './components/AssetExample';

// or any pure javascript modules available in npm
// import { Card } from 'react-native-elements'; // 0.18.5
import Carousel from 'react-native-snap-carousel'; // 3.6.0

import "@expo/vector-icons"; // 6.3.1

const DATA = ['hello', 'world','my','old','friend'];

export default class App extends Component {
  state = {
    index: 0
  }
  
  render() {
    const SLIDER_WIDTH = Dimensions.get("window").width;
    const ITEM_WIDTH = SLIDER_WIDTH - 50;

    return (
      <View >
        <Carousel 
          data={DATA}
          renderItem={({ item }) => {
            return (
              <View style={{
                  backgroundColor: 'blue', 
                  width: ITEM_WIDTH , 
                  height: 200,
                }}
              >
                <Text>{item}</Text>
              </View>
            );
          }}
          sliderWidth={SLIDER_WIDTH}
          itemWidth={ITEM_WIDTH}
          slideStyle={{ width: ITEM_WIDTH }}
          containerCustomStyle={{ marginTop: 50 }}
          inactiveSlideScale={0.97}
          ref={(c) => this.carousel = c}
          onSnapToItem={(index) => this.setState({ index })}
          
        />
        <Text style={{
            marginTop: 20,
            fontSize: 30,
            fontWeight: 'bold',
            textAlign: 'center'
          }}
        >
          {this.state.index}
        </Text>
      </View>
    );
  }
}
`

**Comment by vetrivendhan48** [comment]
I too still face this issue and found a workaround for it.
Just add the following property to your Carousel component
```                              
                  ListFooterComponent={() => <View style={{ minHeight: SOME_HEIGHT,minWidth: REQUIRED_WIDTH }}></View>} // Active index issue fix
```
That's all.
